sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/metrics/_custom_metric.py

```diff
@@ -0,0 +1,165 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import abc
+import importlib
+import json
+import os
+import urllib.parse
+
+from azure.ai.generative.evaluate.metrics._aggregators import mean
+from azure.ai.generative.evaluate.metrics._parsers import ScoreReasonParser, ScoreParser
+
+
+class Metric(metaclass=abc.ABCMeta):
+
+    def __init__(self, name):
+        self.name = name
+
+
+class CodeMetric(Metric):
+    """Evaluation code metric
+
+        :keyword name: Name of the metric.
+        :paramtype name: str
+        :keyword calculate: Function to compute row level metrics. It should have the following signature:
+
+            .. code-block:: python
+
+                def my_code_metric(
+                    *,
+                    data: Dict,
+                    response: Union[Dict, str],
+                    **kwargs,
+                ):
+                    '''
+                        :keyword data: Row data.This method will be called for each row.
+                        :paramtype data: Dict
+                        :keyword response: Response from target function passed to evaluate API otherwise None.
+                        :paramtype response: Dict
+                        :keyword kwargs: Includes params like data_mapping for this metric passed to evaluate API.
+                        :paramtype kwargs: Dict
+                    '''
+                ...
+        :paramtype calculate: Callable
+        :keyword aggregator: Function to aggregate row level metrics. It should have the following signature:
+            .. code-block:: python
+
+                def my_aggregator(
+                    *,
+                    values,
+                    **kwargs,
+                ):
+                    '''
+                        :keyword values: Row level metric value calculated by calculate method of the metric.
+                        :paramtype values: Union[int, Dict[str, int]]
+                        :keyword kwargs: Includes params like data_mapping for this metric passed to evaluate API.
+                        :paramtype kwargs: Dict
+                    '''
+                ...
+        :paramtype aggregator: Callable
+
+    """
+
+    def __init__(self, *, name, calculate, aggregator=None, **kwargs):
+        super(CodeMetric, self).__init__(name=name)
+        self.calculate = calculate
+        self.aggregator = aggregator if aggregator else mean
+
+
+# WIP: This implementation will change
+class PromptMetric(Metric):
```

Comment:
Can you change this code sample to use the official sample syntax? You can use this PR as an example:

https://github.com/Azure/azure-sdk-for-python/pull/32284/files

Also, here's azure-sdk-for-python's sample guide: https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/sample_guide.md#capturing-code-snippets-in-reference-documentation

Reply:
Thanks @diondrapeck  will handle this in a subsequent PR since need to add more samples. Thanks for the comment